### PR TITLE
BAC-3224: Adding AccessPolicyV2 support

### PR DESCRIPTION
**Pull request by burakcive**

### CODE REVIEW

File: internal/provider/access_policies_data_source.go [hunk 1]
```diff
@@ -73,8 +73,50 @@ func (d *accessPoliciesDataSource) Schema(_ context.Context, _ datasource.Schema
 							ElementType: types.StringType,
 						},
 						"credential_provider": schema.StringAttribute{
-							Description: "Credential Provider ID configured in the Access Policy.",
+							Description:        "Configured server workload of the access policy.",
```
Comment (m43kwon) [inline internal/provider/access_policies_data_source.go:76]: Should this description be for the Credential Provider, not the server workload?

File: internal/provider/access_policies_data_source.go [hunk 2]
```diff
@@ -73,8 +73,50 @@ func (d *accessPoliciesDataSource) Schema(_ context.Context, _ datasource.Schema
...
+									"httpbody_field_path": schema.StringAttribute{
+										Description: "Field path in the HTTP body for the credential provider.",
+										Optional:    true,
+									},
+									"httpbody_field_value": schema.StringAttribute{
+										Description: "Field value in the HTTP body for the credential provider.",
+										Optional:    true,
+									},
+									"account_name": schema.StringAttribute{
+										Description: "Name of the snowflake account for the credential provider.",
```
Comment (m43kwon) [inline internal/provider/access_policies_data_source.go:115]: Is this field only for snowflake? And can we capitalize snowflake -> Snowflake?

File: internal/provider/access_policy_resource.go [hunk 3]
```diff
@@ -91,12 +91,55 @@ func (r *accessPolicyResource) Schema(_ context.Context, _ resource.SchemaReques
...
+						"httpbody_field_path": schema.StringAttribute{
+							Description: "Field path in the HTTP body for the credential provider.",
+							Optional:    true,
+						},
+						"httpbody_field_value": schema.StringAttribute{
+							Description: "Field value in the HTTP body for the credential provider.",
+							Optional:    true,
+						},
+						"account_name": schema.StringAttribute{
+							Description: "Name of the snowflake account for the credential provider.",
```
Comment (m43kwon) [inline internal/provider/access_policy_resource.go:137]: Same as above, can we capitalize snowflake -> Snowflake?

File: internal/provider/access_policy_resource.go [hunk 4]
```diff
@@ -261,90 +316,144 @@ func (r *accessPolicyResource) ImportState(ctx context.Context, req resource.Imp
...
+			HeaderValue:          "",
+			HttpbodyFieldPath:    "",
+			HttpbodyFieldValue:   "",
+		}
+	} else {
+		policy.CredentialProviders = make([]aembit.PolicyCredentialMappingDTO, len(model.CredentialProviders))
+
+		for i, credentialProvider := range model.CredentialProviders {
+			policy.CredentialProviders[i] = aembit.PolicyCredentialMappingDTO{
+				PolicyId:             "00000000-0000-0000-0000-000000000000",
```
Comment (m43kwon) [inline internal/provider/access_policy_resource.go:347]: Is PolicyId always this zero value? Is it calculated or generated elsewhere?

File: internal/provider/access_policies_data_source.go (hunk 2 above)
Comment (burakcive) [reply internal/provider/access_policies_data_source.go:115]: As far as I know @quantifex ?

File: internal/provider/access_policy_resource.go (hunk 4 above)
Comment (burakcive) [reply internal/provider/access_policy_resource.go:347]: In create or update API does not use the PolicyId but we cant pass null or empty too. It should be default.
I tried not to send it at all but server return 400.

So I am passing the default GUID value.

File: tests/policy/TestAccBasicAccessPolicyResource.tf [hunk 5]
```diff
@@ -54,14 +77,39 @@ resource "aembit_access_policy" "first_policy" {
...
     name = "TF Second Policy"
     client_workload = aembit_client_workload.second_client.id
-    credential_provider = aembit_credential_provider.api_key.id
+    credential_provider = aembit_credential_provider.snowflake2.id
     server_workload = aembit_server_workload.first_server.id
-}
\ No newline at end of file
+}
+
+resource "aembit_access_policy" "third_policy" {
```
Comment (quantifex) [inline tests/policy/TestAccBasicAccessPolicyResource.tf:92]: Please move the new multiple credential provider resource tests into a standalone tf/tfmod file set.

File: internal/provider/access_policy_resource.go [hunk 6]
```diff
@@ -70,33 +72,89 @@ func (r *accessPolicyResource) Schema(_ context.Context, _ resource.SchemaReques
...
+					"credential_provider_id": types.StringType,
+					"mapping_type":           types.StringType,
+					"header_name":            types.StringType,
+					"header_value":           types.StringType,
+					"httpbody_field_path":    types.StringType,
+					"httpbody_field_value":   types.StringType,
+				}}, []attr.Value{})),
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"policy_id": schema.StringAttribute{
```
Comment (quantifex) [inline internal/provider/access_policy_resource.go:117]: As above, let's get rid of policy_id here, it provides no value

File: internal/provider/access_policy_resource.go [hunk 7]
```diff
@@ -70,33 +72,89 @@ func (r *accessPolicyResource) Schema(_ context.Context, _ resource.SchemaReques
...
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
 				},
 			},
+			"credential_providers": schema.SetNestedAttribute{
+				Description: "Set of Credential Providers to enforce on the Access Policy.",
+				Optional:    true,
+				Computed:    true,
+				Default: setdefault.StaticValue(types.SetValueMust(types.ObjectType{AttrTypes: map[string]attr.Type{
+					"policy_id":              types.StringType,
```
Comment (quantifex) [inline internal/provider/access_policy_resource.go:107]: As above, let's get rid of policy_id here, it provides no value

**Review by quantifex — COMMENTED** [review]
@burakcive Let me know when you've made these small updates and then I'd like to dig into the []string issue a bit further.

**Pull request merged**